### PR TITLE
Fixes from #562

### PR DESCRIPTION
**Pull request by elmarco**

### CODE REVIEW

**Review by zeenix — APPROVED** [review]
LGTM otherwise.

File: zbus/src/connection/socket/unix.rs [hunk 1]
```diff
@@ -26,6 +26,8 @@ use nix::{
 use super::{ReadHalf, RecvmsgResult, WriteHalf};
 #[cfg(feature = "tokio")]
 use super::{Socket, Split};
+
+#[allow(unused_imports)]
```
[the remark this answers is not in the document]
Comment (elmarco) [reply zbus/src/connection/socket/unix.rs:30]: ok, full path then.

File: zbus/src/connection/socket/tcp.rs [hunk 2]
```diff
@@ -102,6 +103,19 @@ impl Socket for tokio::net::TcpStream {
     }
 }
 
+#[cfg(feature = "tokio")]
+#[cfg(windows)]
+fn win32_credentials_from_addr(addr: &std::net::SocketAddr) -> io::Result<ConnectionCredentials> {
```
[the remark this answers is not in the document]
Comment (elmarco) [reply zbus/src/connection/socket/tcp.rs:108]: ok

File: zbus/src/connection/socket/tcp.rs [hunk 3]
```diff
@@ -143,4 +162,9 @@ impl WriteHalf for tokio::net::tcp::OwnedWriteHalf {
     async fn close(&mut self) -> io::Result<()> {
         tokio::io::AsyncWriteExt::shutdown(self).await
     }
+
+    #[cfg(windows)]
+    async fn peer_credentials(&mut self) -> io::Result<ConnectionCredentials> {
+        win32_credentials_from_addr(&self.peer_addr()?)
```
[the remark this answers is not in the document]
Comment (zeenix) [reply zbus/src/connection/socket/tcp.rs:154]: you don't need `self`, you need ` win32_credentials_from_addr`, which is Copy, so you can just pass by value.

**Pull request merged**